src/Data/Component.elm

```diff
@@ -123,6 +124,18 @@ type alias ExpandedElement =
     }
 
 
+{-| Index of an item element and associated source component
+-}
+type alias TargetElement =
+    ( Component, Int, Int )
```

Comment:
Do we agree that it would be the same than `( TargetItem, Int )`? And what about renaming the `Int` type to a custom `Index` type for example (same for the `Int` in `TargetItem` below)?
`(TargetItem, Index)` would make more sense to me than `(Component, Int, Int)` and `(Component, Index)` more sense than `(Component, Int)`, what do you think?